zebra-state/src/service/check.rs

```diff
@@ -229,14 +229,27 @@ fn difficulty_threshold_is_valid(
     let median_time_past = difficulty_adjustment.median_time_past();
     let block_time_max =
         median_time_past + Duration::seconds(difficulty::BLOCK_MAX_TIME_SINCE_MEDIAN);
+
+    // # Consensus
+    //
+    // > For each block other than the genesis block, `nTime` MUST be strictly greater
+    // than the median-time-past of that block.
+    //
+    // https://zips.z.cash/protocol/protocol.pdf#blockheader
     if candidate_time <= median_time_past {
         Err(ValidateContextError::TimeTooEarly {
```

Comment:
It's not clear how "For each block other than the genesis block" is enforced here. (I think it's enforced by the calling function, but we might not want to depend on that.)

Can we change the code so this check always skips the genesis block?

Reply:
https://github.com/ZcashFoundation/zebra/pull/3296/commits/476e96feb45c521b34bb14ed0dea710f4385e1c4